src/javascript/app/pages/cashier/dp2p.js

```diff
@@ -59,29 +66,43 @@ const DP2P = (() => {
                     cursor: pointer;
                 }
                 `;
-                el_main_css.rel = 'stylesheet';
-
-                const dp2p_props = {
-                    className    : 'theme--light',
-                    websocket_api: BinarySocket,
-                    lang         : getLanguage(),
-                    client       : {
-                        currency  : Client.get('currency'),
-                        is_virtual: Client.get('is_virtual'),
-                    },
-                };
-
-                ReactDOM.render(
-                    // eslint-disable-next-line no-console
-                    React.createElement(module.default, dp2p_props),
-                    shadowed_el_dp2p
-                );
-
-                shadowed_el_dp2p.prepend(el_main_css);
-            });
-        } else {
-            document.getElementById('message_cashier_unavailable').setVisibility(1);
-        }
+        el_main_css.rel = 'stylesheet';
+
+        const p2pSubscribe = (request, cb) => {
+            // Request object first key will be the msg_type
+            const msg_type = Object.keys(request)[0];
+
+            SubscriptionManager.subscribe(msg_type, request, cb);
+            return {
+                unsubscribe: () => SubscriptionManager.forget(msg_type),
+            };
+        };
+
+        const websocket_api = {
+            send: BinarySocket.send,
+            wait: BinarySocket.wait,
+            p2pSubscribe,
+        };
+
+        const dp2p_props = {
+            className: 'theme--light',
+            client   : {
+                currency             : Client.get('currency'),
+                is_virtual           : Client.get('is_virtual'),
+                local_currency_config: Client.get('local_currency_config'),
+                residence            : Client.get('residence'),
+            },
+            lang: getLanguage(),
+            websocket_api,
+        };
+
+        ReactDOM.render(
+            // eslint-disable-next-line no-console
```

Comment:
Why we need this line (eslint-disable) in PR?

Reply:
@mahdipourziaei-binary could you remove this line?

Reply:
@negarn It's done. Just waiting for @mustofa-binary to get my commits.

Reply:
its done 👍